### PR TITLE
allow cardboard UI to be disabled (fixes #52)

### LINKED ISSUE #52: Allow Cardboard UI to be disabled
It'd be nice if there was a pref on `WebVRConfig` to not create the Cardboard UI (the back arrow, centerline, gear icon for settings).

### PR DESCRIPTION
**Pull request by cvan**
dependent on #51

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,17 +43,16 @@ are supported:
...
   // Forces availability of VR mode, even for non-mobile devices.
   FORCE_ENABLE_VR: true, // Default: false.
 
   // Complementary filter coefficient. 0 for accelerometer, 1 for gyro.
   K_FILTER: 0.98, // Default: 0.98.
 
-  // How far into the future to predict during fast motion.
-  PREDICTION_TIME_S: 0.050, // Default: 0.050s.
+  // How far into the future to predict during fast motion (in seconds).
+  PREDICTION_TIME_S: 0.040, // Default: 0.040.
```
Comment (borismus) [inline README.md]: Why is this changed?

File: src/cardboard-distorter.js [hunk 2]
```diff
@@ -53,7 +53,7 @@ function CardboardDistorter(gl) {
   this.meshWidth = 20;
   this.meshHeight = 20;
 
-  this.bufferScale = WebVRConfig.BUFFER_SCALE ? WebVRConfig.BUFFER_SCALE : 1.0;
```
Comment (borismus) [inline src/cardboard-distorter.js]: Not safe.

File: src/cardboard-ui.js [hunk 3]
```diff
@@ -137,6 +138,8 @@ CardboardUI.prototype.listen = function(optionsCallback, backCallback) {
  * Builds the UI mesh.
  */
 CardboardUI.prototype.onResize = function() {
+  if (!WebVRConfig.ENABLE_CARDBOARD_UI) return false;
```
Comment (borismus) [inline src/cardboard-ui.js]: Cardboard UI should be enabled by default. The flag should be DISABLE_CARDBOARD_UI.

File: build/webvr-polyfill.js [hunk 4]
```diff
@@ -4274,20 +4371,64 @@ module.exports = Emitter;
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+var extend = _dereq_('./util.js').extend;
```
Comment (borismus) [inline build/webvr-polyfill.js]: Let's be consistent. Include all of Util and use Util.extend.

File: src/util.js [hunk 5]
```diff
@@ -179,5 +179,25 @@ Util.isMobile = function() {
   return check;
 };
 
+// Adapted from https://mdn.io/Object/assign#Polyfill
```
Comment (borismus) [inline src/util.js]: Why not call it Object.assign then?

File: README.md (hunk 1 above)
Comment (cvan) [inline README.md]: I noticed that the default here was wrong - it's actually [`0.040`](https://github.com/borismus/webvr-polyfill/pull/53/files#diff-c9f2bc69ace9478cfd1156aabd2a8e31L4814)

File: build/webvr-polyfill.js (hunk 4 above)
Comment (cvan) [inline build/webvr-polyfill.js]: sure thing, done

File: src/cardboard-ui.js (hunk 3 above)
Comment (cvan) [inline src/cardboard-ui.js]: that'll [happen now](https://github.com/borismus/webvr-polyfill/pull/53/files#diff-1c90ff38a08209f9ebd4d05d1e43358eR48) with the default config object I added to `src/main`

Comment (cvan) [inline src/cardboard-ui.js]: handled in PR #51 - included this commit but will remove from this PR when it gets merged

File: src/util.js (hunk 5 above)
Comment (cvan) [inline src/util.js]: it's not natively supported in any version of WebKit and not in IE 11 (but in Edge, yes): http://kangax.github.io/compat-table/es6/#test-Object_static_methods_Object.assign

**Comment by borismus** [comment]
I'd like to merge this, but seems to have diverged from master. Could you rebase please?

**Comment by borismus** [comment]
@cvan Sorry for delays, was busy with VR Views, etc. Could you revisit these conflicts so we can merge?

**Comment by cvan** [comment]
np, updated

File: src/webvr-polyfill.js [hunk 6]
```diff
@@ -37,7 +37,7 @@ function WebVRPolyfill() {
     if (!this.nativeWebVRAvailable) {
       this.enablePolyfill();
     }
-    if (WebVRConfig.ENABLE_DEPRECATED_API) {
```
Comment (borismus) [inline src/webvr-polyfill.js]: Why the rename here?

File: src/cardboard-vr-display.js [hunk 7]
```diff
@@ -128,13 +128,15 @@ CardboardVRDisplay.prototype.beginPresent_ = function() {
...
 
-  this.cardboardUI_.listen(function() {
-    // Options clicked
-    this.viewerSelector_.show(this.layer_.source.parentElement);
-  }.bind(this), function() {
-    // Back clicked
-    this.exitPresent();
-  }.bind(this));
+  if (WebVRConfig.CARDBOARD_UI) {
+    this.cardboardUI_.listen(function() {
```
Comment (borismus) [inline src/cardboard-vr-display.js]: This will just make the buttons not work, don't think it will hide the UI.

**Comment by borismus** [comment]
Generally speaking, what are you trying to achieve? The cardboard UI is: the vertical line, back button and settings gear. When do you not want those things?

**Comment by borismus** [comment]
Bump

**Comment by revolunet** [comment]
in some case we know what settings to apply so the settings can be useless (ex: custom installations)
